packages/astro/src/core/errors/dev/utils.ts

```diff
@@ -32,10 +33,15 @@ export function collectErrorMetadata(e: any, rootFolder?: URL | undefined): Erro
 		// - We'll fail to show the file's content in the browser
 		// - We'll fail to show the code frame in the terminal
 		// - The "Open in Editor" button won't work
+
+		// Normalize the paths so that we can correctly detect if it's absolute on any platform
+		const normalizedFile = error.loc?.file?.replace(/\\/g, '/');
+		const normalizedRootFolder = removeLeadingForwardSlashWindows(rootFolder?.pathname || '');
```

Comment:
Remove a leading forward slash which `rootFolder.pathname` comes with which is invalid on Windows (e.d. `/c:/`) and will always fail the check for the path being absolute below which results in the bug that is described in the issue.